### ISSUE TITLE
Amplify API REST Delete executes a GET instead of DELETE method when there is no body

### ISSUE THREAD

**Issue by Thomas-Boutin** [aws-amplify/amplify-android#1932]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin, Kotlin - Coroutines

### Amplify Categories

REST API

### Gradle script dependencies

<details>

```groovy

def amplifyKotlinCoreVersion = '0.21.2'
def amplifyframeworkVersion = '1.37.2'
implementation "com.amplifyframework:aws-auth-cognito:$amplifyframeworkVersion"
implementation "com.amplifyframework:aws-api:$amplifyframeworkVersion"
implementation "com.amplifyframework:core-kotlin:$amplifyKotlinCoreVersion"


```

</details>

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

Hi !

On Android, an Amplify REST DELETE call is turned into a GET if the body is null.
Considering that the classes responsible of the bug are not written in Kotlin, the Java version should be impacted too.

I've looked further and managed to find where the bug is located : we put the HTTP method type in the strategy passed to **RestRequestFactory**.

![image](https://user-images.githubusercontent.com/10796070/188196915-4651e422-8698-404a-a984-4ff7c2ab759f.png)
![image](https://user-images.githubusercontent.com/10796070/188197097-bff6fa2b-a5b0-46b5-9f8c-9e1bb51fc46f.png)

However, as you can see, if **data** is null, the strategy is not executed.

As a result, the generated request is a GET instead of a DELETE :
<img width="666" alt="delete-1" src="https://user-images.githubusercontent.com/10796070/188197257-2d22a44c-0303-4f53-a189-b5fa59a82628.png">

Modifying the call like to pass an empty body is a functional workaround :
```kotlin
val options: RestOptions = RestOptions.builder()
    .addPath("$GET_TRIPS_URL/$id")
    .addBody(ByteArray(0))
    .build()

val response = Amplify.API.delete(options)
```

### Reproduction steps (if applicable)

- Through API gateway, create a DELETE Rest endpoint which deletes an item with an id passed as a path parameter
- On Android, try to delete it by calling DELETE without passing a body in the query

### Code Snippet

```kotlin
val options: RestOptions = RestOptions.builder()
    .addPath("$ENDPOINT/$id")
    .build()

val response = Amplify.API.delete(options)

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by ElementalSystems** [comment]
My team has also experienced this - workaround (zero length body) works perfectly.

**Comment by eeatonaws** [comment]
A fix has been released in Amplify Android 2.8.0 and 1.38.5. Please let us know if you have any additional questions.

**Comment by sekitaka** [comment]
I updated to Amplify Android 2.8.5, then it works fine. Thank you.
